frontend/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -522,10 +539,44 @@ function* applyStateOfStack(
 function mergeDataWithBackendDataInPlace(
   originalData: BucketDataArray,
   backendData: BucketDataArray,
+  pendingOperations: Array<(BucketDataArray) => void>,
+  log: boolean,
 ) {
-  for (let i = 0; i < originalData.length; ++i) {
-    originalData[i] = originalData[i] || backendData[i];
+  if (originalData.length !== backendData.length) {
```

Comment:
The old code occasionally merged uint8 with uint32 data since the decompression always returned uint8.